### ISSUE TITLE
Associations with old data are not removed/updated.

### ISSUE THREAD

**Issue by faultyserver** [propershark/shark#24]
Vehicles are continuously sending `depart` events to stations that they left long ago, meaning that `dissociate` is either not working as expected (or is not being called appropriately), or the `last_station` property for the vehicles is not being properly.

**Comment by faultyserver** [comment]
Testing has shown that `dissociate` is indeed working properly, so there is most likely an issue in `Conductor` that is keeping the method from being called appropriately.

**Comment by faultyserver** [comment]
After lots of confusion over how `dissociate` could be working, but somehow returning `true` when the association shouldn't have existed, I think I've finally determined why this is happening.

For CityBus, `vehicle.last_station` and `vehicle.next_station` are sourced different locations; `last_station` comes from DoubleMap, and `next_station` comes from TripSpark. It is perfectly reasonable to assume that these sources can be out-of-sync at any time. If DoubleMap updates before TripSpark does (which is pretty common), `last_station` will be the same as `next_station` until TripSpark updates appropriately.

So, the issue here is not that the associations are not removed, but rather that the associations are being removed and re-added in update cycles where the two sources are out-of-sync.

The solution that I've come up with is to change Conductor's `vehicles:update` handler to something like this:

``` ruby
if dissociate(last_station, vehicle)
  fire(:depart)
end
unless next_station == last_station
  associate(next_station, vehicle)
end
```

Having the dissociation come first ensures that the `depart` event is sent as soon as `last_station` changes. Then, the guard clause around `associate` means that the `next_station` is only associated if it is not the same as the `last_station`, ensuring that the sources are synchronized.

Will test and report back.

**Comment by faultyserver** [comment]
Closed by 54cb98a. The guard clause given above seems to have worked properly.
